### PR TITLE
docs(architecture): привязать неизменяемость ADR к целевой ветке

### PR DESCRIPTION
**Pull request by zeegin**
## Что меняется

Закрывает #269. `INV-DOC-SUPERSEDE-NOT-EDIT` связывал неизменяемость записи со
словом `accepted` и не называл границу целевой ветки, поэтому одну и ту же
правку ADR внутри незавершённого PR можно было прочитать и как нарушение, и как
обычную работу по ревью — расхождение, всплывшее на ревью #265 при
одновременных номерах в #265 и #266. Теперь жизненный цикл сформулирован один
раз и одинаково в инварианте, `spec/decisions/README.md`, `spec/README.md`,
чек-листе изменений и `AGENTS.md`: запись становится историей в момент
попадания в целевую ветку (обычный поток — `main`, ветка задаётся проверке
явно, поэтому релизный поток с другой base-веткой не ломается); до слияния её
правят, перенумеровывают, объединяют и удаляют внутри своего PR, после слияния
— замещают, а не переписывают. Статус `superseded` получил происхождение: это
переход из `accepted` в целевой ветке, поэтому запись не может впервые попасть
в каталог уже замещённой и не может прыгнуть в `superseded` из `proposed`.
Заменённую до слияния запись удаляют, переписывают или объединяют с
заменяющей.

Страж `scripts/ci/check-architecture-sync.py` больше не судит по одному diff: он
читает состав `spec/decisions/` на вершине целевой ветки. Отсюда четыре
следствия — запись, которой в целевой ветке нет, под правила неизменяемости не
попадает; целевая ветка подставляет `before`-статус, которого нет в diff;
переход в `superseded` обязан назвать заменяющую запись разрешимой ссылкой
(`ADR-NNNN` или ссылка на файл); номер, уже израсходованный в целевой ветке,
новой записью не занимается. Без читаемой base-ветки страж возвращается к
прежнему diff-only чтению — более строгому из двух, а не более мягкому.

## Архитектурный слой

- Затронутые записи реестра: `INV-DOC-SUPERSEDE-NOT-EDIT` (переформулировано
  поле `Rule`).
- Решение (ADR), если публичный или архитектурный контракт меняется: не
  требуется. Поверхность `unica.*`, идентичность MCP-сервера, упаковка и
  границы слоёв не тронуты, а у записи `INV-DOC-SUPERSEDE-NOT-EDIT` поле
  `Decision` — `n/a`: правило процесса документации своего ADR-владельца не
  имеет.

- [x] Пройден [чек-лист изменений](../spec/architecture/change-checklist.md)
      в части, относящейся к этому изменению.

## Проверка

```sh
python3.12 -m unittest tests.ci.test_architecture_sync_guard tests.ci.test_architecture_registry
python3.12 -m unittest discover -s tests/ci
python3.12 scripts/ci/check-architecture-sync.py --base origin/main --strict
git diff --check
```

`tests/ci/test_architecture_sync_guard.py` вырос с 63 до 84 тестов. Новое
поведение покрывают классы `TargetBranchLifecycleTests` (правка,
перенумерация и удаление ещё не влитой записи; неизменяемость влитой;
`superseded` у не влитой записи; занятый номер; статус из целевой ветки как
`before`) и `BaseCatalogueReadTests` (чтение каталога из ветки, откат к
diff-only чтению, base-ветка для diff со stdin), плюс новые тесты перехода в
`superseded` в `DecisionRecordImmutabilityTests`. Кроме тестов страж прогнан
на реальных коммитах: правка не влитой ADR-0020 (в том числе в виде
per-commit diff, как на ревью #265) проходит, она же со статусом `superseded`
падает, правка даты влитой ADR-0011 падает, замещение ADR-0011 со ссылкой
проходит, без ссылки — падает.

`cargo fmt`/`clippy`/`test` не запускались: Rust в этом изменении не тронут.
Полный `unittest discover -s tests/ci` даёт 11 локальных падений в
`test_reference_format_profile` из-за отсутствующего `lxml` — тот же базовый
набор, что и на `main`, к этому изменению отношения не имеет.

## Неграницы

Фактическая очистка и перенумерация ADR в #265 и #266 остаётся этим PR не
тронутой: правило принимается здесь, применяется — там.

## Summary by CodeRabbit

* **Documentation**
  * Reworded and expanded ADR immutability rules, including the exact boundary when records become immutable after landing in the target branch.
  * Clarified `accepted` vs `superseded`, replacement referencing requirements, and numbering behavior for parallel pull requests.

* **Bug Fixes**
  * Updated the architecture synchronization guard to analyze decision records against the target branch’s existing catalogue (base-aware).
  * Strengthened validation for superseding transitions and replacement identification.

* **Tests**
  * Expanded test coverage for lifecycle and base-aware scenarios, including status-link/replacement naming rules and edge cases.

### CODE REVIEW

**Comment by zeegin** [comment]
Разобрал три замечания ревью (все три пришли только в теле ревью, без inline-веток, поэтому отвечаю одним комментарием).

- `tests/ci/test_architecture_sync_guard.py` — принято, `d57f260`. Тест фиксировал `status("0011") == "accepted"`, то есть именно то значение, которое этот PR разрешает менять; теперь проверяется, что распознан известный статус, а не какой именно. Проверки `has("0011")` и `has("0007")` остались: файл записи из каталога не уходит, а выведенный номер не возвращается — обе стороны стабильны.
- `scripts/ci/check-architecture-sync.py`, `DECISION_RECORD` из `DECISIONS_DIR` — принято, `d57f260`. Две записи одного и того же префикса рядом друг с другом — это ровно тот дрейф, от которого страж защищает сам себя разделом про переехавшие якоря.
- `scripts/ci/check-architecture-sync.py`, один `git cat-file --batch` вместо `git show` на запись — не делаю. Каталог сейчас 19 файлов, это ~19 быстрых процессов в задании, которое и так идёт минуты; выигрыш не измеряется, а протокол `--batch` со stdin/stdout добавляет разбор, который в этом файле ценится ниже читаемости. Вернуться сюда стоит, когда каталог вырастет на порядок.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: scripts/ci/check-architecture-sync.py [hunk 1]
```diff
@@ -398,40 +428,129 @@ def unreadable_sections(diff_text: str) -> list[str]:
...
+        content = subprocess.run(
+            ["git", "show", f"{base}:{path}"],
+            cwd=REPO_ROOT,
+            capture_output=True,
+            text=True,
+        )
+        if content.returncode != 0:
+            statuses[identifier] = None
+            continue
+        statuses[identifier] = base_status(content.stdout)
```
[the remark this answers is not in the document]
Comment (zeegin) [reply scripts/ci/check-architecture-sync.py:495]: Принято, `13dfa9a`: оба чтения теперь идут с `encoding="utf-8", errors="replace"`, а в docstring записано, почему кодировка не отдаётся локали. Покрыто тестом `test_the_reads_pin_utf_8_instead_of_trusting_the_locale`. Тот же паттерн есть в `read_diff` — он существовал до этого PR и выносится отдельно, чтобы не расширять границу этого изменения.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
